Modules/IO/NIFTI/test/itkNiftiImageIOTest14.cxx

```diff
@@ -0,0 +1,104 @@
+/*=========================================================================
+ *
+ *  Copyright NumFOCUS
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *         https://www.apache.org/licenses/LICENSE-2.0.txt
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ *=========================================================================*/
+
+#include "itkNiftiImageIOTest.h"
+#include "itkTestingMacros.h"
+
+// Test writing NIFTI xyzt_units field - should be 10 (mm, sec)
+int
+itkNiftiImageIOTest14(int argc, char * argv[])
+{
+
+  if (argc > 1)
+  {
+    char * testdir = *++argv;
+    itksys::SystemTools::ChangeDirectory(testdir);
+  }
+  else
+  {
+    return EXIT_FAILURE;
+  }
+  int success(0);
+
+  constexpr unsigned int Dimension = 3;
+
+  using PixelType = float;
+  using ImageType = itk::Image<PixelType, Dimension>;
+
+  ImageType::Pointer image;
+
+  // fill out an image and write it as nifti.
+  const char * filename = "xyzt_unit_check.nii";
+  using OutputImageType = itk::Image<PixelType, Dimension>;
+  typename OutputImageType::RegionType region;
+  typename OutputImageType::IndexType  start;
+  start[0] = 0;
+  start[1] = 0;
+  start[2] = 0;
+  typename OutputImageType::SizeType size;
+  size[0] = 8;
+  size[1] = 8;
+  size[2] = 4;
+  region.SetSize(size);
+  region.SetIndex(start);
+  auto outputimage = OutputImageType::New();
+  outputimage->SetRegions(region);
+  outputimage->Allocate();
+
+  using WriterType = itk::ImageFileWriter<OutputImageType>;
+  auto                       writer = WriterType::New();
+  itk::NiftiImageIO::Pointer niftiImageIO(itk::NiftiImageIO::New());
```

Comment:
Just like `auto writer = WriterType::New()`, you may of course also do `auto niftiImageIO = itk::NiftiImageIO::New()`. Just a suggestion.